conans/client/graph/compute_pid.py

```diff
@@ -78,3 +74,8 @@ def run_package_id(conanfile):
                     conanfile.info.invalid = BINARY_INVALID, str(e)
                 except ConanErrorConfiguration as e:
                     conanfile.info.invalid = BINARY_ERROR, str(e)
+
+    # Once we are done, call package_id() to narrow and change possible values
+    with conanfile_exception_formatter(conanfile, "package_id"):
```

Comment:
I have tested this order change (1.X package_id + validate, 2.0 validate + package_id) in Conan 1.X, it only breaks 1 test using ``compatible_packages``, which would be fixed if we backport the ``compatibility()`` method